source/gboardforensics/gatherers/trainingcache.d

```diff
@@ -0,0 +1,153 @@
+module gboardforensics.gatherers.trainingcache;
+
+import gboardforensics.gatherers;
+import gboardforensics.models.trainingcache;
+import gboardforensics.analysis : DB;
+
+import std.algorithm;
+import std.array;
+import std.conv;
+import std.file;
+import std.path;
+import std.range : ElementType;
+
+import core.thread;
+
+import d2sqlite3;
+
+class TrainingCacheGatherer : IGatherer
+{
+	this(string path)
+		in (path.exists())
+		in (path.isFile())
+	{
+		_trainingcache.path = path;
+		_db = cast(DB) path.baseName();
+	}
+
+	void gather()
+	{
+		Database db = Database(_trainingcache.path);
+		scope(exit) db.close();
+
+		switch (_db)
```

Comment:
This can be optimized by passing an integer enum to the constructor and match that instead. See this example: https://godbolt.org/z/oEe7h9av3

Reply:
done